src/input/input_python.rs

```diff
@@ -109,12 +109,12 @@ impl<'a> Input<'a> for PyAny {
         self.is_none()
     }
 
-    fn input_get_attr(&self, name: &PyString) -> Option<PyResult<&PyAny>> {
-        Some(self.getattr(name))
-    }
-
-    fn is_exact_instance(&self, class: &PyType) -> bool {
-        self.get_type().is(class)
+    fn input_is_instance(&self, class: &PyType) -> Option<&PyAny> {
```

Comment:
Not a big deal, but given this returns an `Option<&PyAny>` rather than a `bool` the name seems a bit weird to me (I would expect it to return a `bool` since the name sounds like an assertion). I'd suggest `downcast` or similar, but  also fine keeping as is if you prefer.